drivers/dma/dma_andes_atcdmac300.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2023 Andes Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_DMA_ANDES_ATCDMAC300_H_
+#define ZEPHYR_DRIVERS_DMA_ANDES_ATCDMAC300_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define ATCDMAC100_MAX_CHAN	8
+
+typedef void (*atcdmac300_cfg_func_t)(void);
+
+struct chain_block {
+	uint32_t ctrl;
+	uint32_t transize;
+	uint32_t srcaddrl;
+	uint32_t srcaddrh;
+	uint32_t dstaddrl;
+	uint32_t dstaddrh;
+	uint32_t llpointerl;
+	uint32_t llpointerh;
+	struct chain_block *next_block;
+};
+
+/* data for each DMA channel */
+struct dma_chan_data {
+	void *blkuser_data;
+	dma_callback_t blkcallback;
+	struct chain_block *head_block;
+	struct dma_status status;
+	struct k_spinlock lock;
+};
+
+/* Device run time data */
+struct dma_atcdmac300_data {
+	struct dma_chan_data chan[ATCDMAC100_MAX_CHAN];
+};
+
+/* Device constant configuration parameters */
+struct dma_atcdmac300_cfg {
+	atcdmac300_cfg_func_t irq_config;
+	uint32_t base;
+	uint32_t irq_num;
+};
+
+#define DEV_CFG(dev) ((const struct dma_atcdmac300_cfg * const) (dev)->config)
+#define DEV_DATA(dev) ((struct dma_atcdmac300_data *) (dev)->data)
+
+#define DMA_IDR (DEV_CFG(dev)->base + 0x00)
+#define DMA_CFG (DEV_CFG(dev)->base + 0x10)
+#define DMA_CTRL (DEV_CFG(dev)->base + 0x20)
+#define DMA_ABORT (DEV_CFG(dev)->base + 0x24)
+#define DMA_INT_STATUS (DEV_CFG(dev)->base + 0x30)
+#define DMA_CH_ENABLE (DEV_CFG(dev)->base + 0x34)
+
+#define DMA_CH_OFFSET(ch) (ch * 0x20)
+#define DMA_CH_CTRL(ch) (DEV_CFG(dev)->base + 0x40 + DMA_CH_OFFSET(ch))
+#define DMA_CH_TRANSIZE(ch) (DEV_CFG(dev)->base + 0x44 + DMA_CH_OFFSET(ch))
+#define DMA_CH_SRC_ADDR_L(ch) (DEV_CFG(dev)->base + 0x48 + DMA_CH_OFFSET(ch))
+#define DMA_CH_SRC_ADDR_H(ch) (DEV_CFG(dev)->base + 0x4C + DMA_CH_OFFSET(ch))
+#define DMA_CH_DST_ADDR_L(ch) (DEV_CFG(dev)->base + 0x50 + DMA_CH_OFFSET(ch))
+#define DMA_CH_DST_ADDR_H(ch) (DEV_CFG(dev)->base + 0x54 + DMA_CH_OFFSET(ch))
+#define DMA_CH_LL_PTR_L(ch) (DEV_CFG(dev)->base + 0x58 + DMA_CH_OFFSET(ch))
+#define DMA_CH_LL_PTR_H(ch) (DEV_CFG(dev)->base + 0x5C + DMA_CH_OFFSET(ch))
```

Comment:
add an explicit `(dev)` parameter to all these macros.

Reply:
Sorry for that we didn't notice that the macro like  DEV_CFG is redundant now.
We have done the modification for it.